Branch-SDK/src/main/java/io/branch/referral/ServerRequest.java

```diff
@@ -451,9 +467,11 @@ void updateGAdsParams() {
         int LATVal = DeviceInfo.getInstance().getSystemObserver().getLATVal();
         String gaid = DeviceInfo.getInstance().getSystemObserver().getAID();
         if (!TextUtils.isEmpty(gaid)) {
-            updateAdvertisingIdsObject(gaid);
-            // gaid is put in the request body below, calling to remove hardware id from request now
-            replaceHardwareIdOnValidAdvertisingId();
+            if (prefHelper_.getConsumerProtectionAttributionLevel() == Defines.BranchAttributionLevel.FULL || !prefHelper_.isAttributionLevelInitialized()) {
```

Comment:
Nit: I think it would be slightly easier to read `&& prefHelper_.isAttributionLevelInitialized()`

I would also suggest that this logical check just be made into its own function since it's repeated several times.

Reply:
So i'm actually checking for either of them to be true, not both. More details in the comment below, but basically the CPP level not being set should be treated the same as it being set to full.